### PR TITLE
Add error for bare list() and dict() calls

### LINKED ISSUE #49: Should recommend list() be replaced with []
test.py

```python
l = list()
print(l)
```

```
$ flake8 test.py
```

Actual: (no output)

Expected: I think perhaps this should output that `list()` is better written as `[]`. Same is true for `dict()` -> `{}`.

I see this as being very similar to "Unnecessary generator - rewrite as a list comprehension.", except it is a literal instead of a comprehension.

What do you think?

Noticed while running against the Django codebase:

https://github.com/django/django/blob/f3427ccb25806bb062becad0d6a0904930b36635/tests/model_inheritance/test_abstract_inheritance.py#L321

### PR DESCRIPTION
**Pull request by jdufresne**
Fixes #49

### CODE REVIEW

**Comment by jdufresne** [comment]
Rebased to resolve merge conflicts.

File: tests/test_it.py [hunk 1]
```diff
@@ -328,3 +328,51 @@ def test_it_does_not_crash_on_attribute_functions():
...
+
+def test_C408_fail_3():
+    errors = run_flake8('dict()')
+    assert errors == [
+        "example.py:1:1: C408 Unnecessary dict call - rewrite as dict literal."
+    ]
+
+
+def test_C408_fail_4():
+    errors = run_flake8('dict(a=1)')
```
Comment (adamchainz) [inline tests/test_it.py]: I don't see how this is triggered if `n_args` is 0? Is there a count of keyword args too?

Comment (jdufresne) [inline tests/test_it.py]: Yes, exactly: `kwonlyargs`. There are more useful details [here](https://greentreesnakes.readthedocs.io/en/latest/nodes.html#function-and-class-definitions).

Comment (adamchainz) [inline tests/test_it.py]: Thanks, that's a useful link

**Comment by adamchainz** [comment]
Merged in 0aa191e861fef80a3eb88747a7a5595a94dbf216, thanks!

**Pull request closed without merging**